### PR TITLE
Playlist: cleanup object store implementation

### PR DESCRIPTION
**Pull request by ryantxu**
In https://github.com/grafana/grafana/pull/58018, I am running into a locked database when trying to use this with SQL.

This PR moves the playlist cleanup out of the SQL implementation, and adds a spinner so we can see when something is submitting

### CODE REVIEW

File: pkg/services/export/object_store.go [hunk 1]
```diff
@@ -261,7 +263,7 @@ func (e *objectStoreJob) getDashboards(ctx context.Context) ([]dashInfo, error)
 	e.broadcaster(e.status)
 
 	dash := make([]dashInfo, 0)
-	rows, err := e.sess.Query(ctx, "SELECT org_id,uid,data,updated_by FROM dashboard WHERE is_folder=0")
+	rows, err := e.sess.Query(ctx, "SELECT org_id,uid,data,updated_by FROM dashboard WHERE is_folder=false")
```
Comment (ryantxu) [inline pkg/services/export/object_store.go:266]: mysql failed with the boolean vs number conversion

File: pkg/services/playlist/playlistimpl/object_store.go [hunk 2]
```diff
@@ -27,30 +27,28 @@ type objectStoreImpl struct {
 var _ playlist.Service = &objectStoreImpl{}
 
 func (s *objectStoreImpl) sync() {
-	rows, err := s.sess.Query(context.Background(), "SELECT org_id,uid FROM playlist ORDER BY org_id asc")
+	type Info struct {
+		OrgID int64  `db:"org_id"`
+		UID   string `db:"uid"`
+	}
+	results := []Info{}
+	err := s.sess.Select(context.Background(), &results, "SELECT org_id,uid FROM playlist ORDER BY org_id asc")
```
Comment (ryantxu) [inline pkg/services/playlist/playlistimpl/object_store.go:35]: rather than processing results within the SQL results loop, this pulls them into a struct first

File: public/app/features/playlist/PlaylistForm.tsx [hunk 3]
```diff
@@ -18,16 +18,22 @@ interface Props {
...
+  const [saving, setSaving] = useState(false);
   const { name, interval, items: propItems } = playlist;
   const tagOptions = useMemo(() => {
     return () => getGrafanaSearcher().tags({ kind: ['dashboard'] });
   }, []);
 
   const { items, addById, addByTag, deleteItem, moveItem } = usePlaylistItems(propItems);
 
+  const doSubmit = (list: Playlist) => {
+    setSaving(true);
```
Comment (nmarrs) [inline public/app/features/playlist/PlaylistForm.tsx:30]: Do we ever set saving to false? May not be necessary though as in testing locally it appears on both success and error user is redirected to main playlist page

File: pkg/services/store/auth.go [hunk 4]
```diff
@@ -44,6 +44,12 @@ func GetUserIDString(user *user.SignedInUser) string {
 	if user == nil {
 		return ""
 	}
+	if user.IsAnonymous {
+		return "anon"
+	}
+	if user.ApiKeyID > 0 {
+		return fmt.Sprintf("key:%d", user.UserID)
+	}
```
Comment (nmarrs) [inline pkg/services/store/auth.go:52]: Are these changes intentional / desired for this PR? Didn't see any usage of `GetUserIDString` func in other backend changes

**Review by nmarrs — APPROVED** [review]
frontend LGTM

File: public/app/features/playlist/PlaylistForm.tsx (hunk 3 above)
Comment (ryantxu) [reply public/app/features/playlist/PlaylistForm.tsx:30]: the parent `onChange` always goes back to the root 🤷

File: pkg/services/store/auth.go (hunk 4 above)
Comment (ryantxu) [reply pkg/services/store/auth.go:52]: ya -- the behavior is only seen depending on which key you use to save objects.  This is minor enough i don't mind adding it here
